### PR TITLE
Profile test

### PR DESCRIPTION
**Pull request by laoliu97**
继续为op增加profile性能测试

### CODE REVIEW

File: python/oneflow/test/modules/test_consistent_broadcast_matmul.py [hunk 1]
```diff
@@ -89,6 +89,12 @@ def test_consistent_broadcast_matmul_with_same_dims(test_case):
                         test_case, placement, x_sbp, y_sbp
                     )
 
+    @profile(torch.matmul)
+    def profile_matmul(test_case):
+        input1 = torch.ones(32,64)
+        input2 = torch.ones(64,128)
+        torch.matmul(input1,input2)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_consistent_broadcast_matmul.py:96]: 文件名带 consistent 的是和多卡相关的，profile 函数应该放在算子本身的测试文件里

File: python/oneflow/test/modules/test_conv2d.py [hunk 2]
```diff
@@ -1590,6 +1590,12 @@ def test_nn_functional_conv2d(test_case):
         y = torch.nn.functional.conv2d(img, kernel, groups=3)
         return y
 
+    @profile(torch.nn.functional.conv2d)
+    def profile_functional_conv2d(test_case):
+        img = torch.ones(8, 4, 3, 3)
+        kernel = torch.ones(1, 4, 5, 5)
+        torch.nn.functional.conv2d(img, kernel, padding=1)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_conv2d.py:1597]: 函数名叫 profile_conv2d 就行，以及需要多添加几组测试用例，尽量覆盖所有的参数，conv1d 也是同样

File: python/oneflow/test/modules/test_consistent_broadcast_matmul.py (hunk 1 above)
Comment (laoliu97) [reply python/oneflow/test/modules/test_consistent_broadcast_matmul.py]: 已修改

File: python/oneflow/test/modules/test_conv2d.py (hunk 2 above)
Comment (laoliu97) [reply python/oneflow/test/modules/test_conv2d.py]: 已修改

File: python/oneflow/test/modules/test_as_stride.py [hunk 3]
```diff
@@ -82,6 +82,13 @@ def test_flow_as_strided_bool(test_case):
         z = torch.as_strided(x, (2, 2, 3), (1, 1, 2), storage_offset)
         return z
 
+    @profile(torch.as_strided) 
+    def profile_as_strided(test_case):
+        input = torch.ones(2,3,5)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_as_stride.py:87]: 输入规模太小了，一般得有 1e4~1e6 个元素才符合实际场景。其它 op 的测试也可以检查下

Comment (laoliu97) [reply python/oneflow/test/modules/test_as_stride.py:87]: 好的

**Comment by laoliu97** [comment]
## op性能测试结果预览
image
image
image
image
image

**Comment by laoliu97** [comment]
## 接口覆盖统计情况预览
image

File: python/oneflow/test/modules/test_conv2d.py [hunk 4]
```diff
@@ -1590,6 +1590,17 @@ def test_nn_functional_conv2d(test_case):
         y = torch.nn.functional.conv2d(img, kernel, groups=3)
         return y
 
+    @profile(torch.nn.functional.conv2d)
+    def profile_conv2d(test_case):
+        inputs = torch.ones(8, 16, 128, 128)
+        filters = torch.ones(64, 16, 3, 3)
+        filters2 = torch.ones(64, 4, 3, 3)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_conv2d.py:1597]:
```suggestion
        filters_16c = torch.ones(64, 16, 3, 3)
        filters_4c = torch.ones(64, 4, 3, 3)
```

File: python/oneflow/test/modules/test_conv2d.py [hunk 5]
```diff
@@ -1590,6 +1590,17 @@ def test_nn_functional_conv2d(test_case):
...
+        torch.nn.functional.conv2d(inputs, filters, padding=1)
+        torch.nn.functional.conv2d(inputs , filters2, groups=4)
+        torch.nn.functional.conv2d(inputs , filters2, bias=torch.ones(64), groups=4)
+        torch.nn.functional.conv2d(inputs , filters2, bias=torch.ones(64),groups=4,stride=4)
+        torch.nn.functional.conv2d(inputs , filters2, bias=torch.ones(64),groups=4,padding=1)
```
[the remark this answers is not in the document]
Comment (laoliu97) [reply python/oneflow/test/modules/test_conv2d.py:1602]: 已修改

File: python/oneflow/test/modules/test_conv1d.py [hunk 6]
```diff
@@ -443,6 +443,31 @@ def test_nn_functional_conv1d(test_case):
         y = torch.nn.functional.conv1d(img, kernel, groups=3)
         return y
 
+    @profile(torch.nn.functional.conv1d)
+    def profile_conv1d(test_case):
+        inputs  = torch.ones(40, 16, 30)
+        filter_16c  = torch.ones(20, 16, 5)
+        filter_4c  = torch.ones(20, 4, 5)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_conv1d.py:450]: 唔，我当时想的 16c 的含义是 output channel 为 16，这里 group=4，output channel 其实还是 16 而不是 4，命名改为 16c_4g 好一些，以及应该和 conv2d 那里统一一下，都用 weight 而不是 filter

File: python/oneflow/test/modules/test_conv2d.py [hunk 7]
```diff
@@ -1999,12 +2000,24 @@ def test_conv2d_NHWC_with_random_data(test_case):
     @profile(torch.nn.functional.conv2d)
     def profile_conv2d(test_case):
         input = torch.ones(8, 128, 28, 28)
-        weight = torch.ones(128, 128, 3, 3)
+        weight_128c = torch.ones(128, 128, 3, 3)
+        weight_64c = torch.ones(128, 64, 3, 3)
+        weight_1X1 = torch.ones(128, 128, 1, 1)
+        weight_5X5 = torch.ones(128, 128, 5, 5)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_conv2d.py:2006]: weight_5x5_128c

Comment (laoliu97) [reply python/oneflow/test/modules/test_conv2d.py]: 已修改

File: python/oneflow/test/modules/test_conv1d.py (hunk 6 above)
Comment (laoliu97) [reply python/oneflow/test/modules/test_conv1d.py]: 已修改

File: python/oneflow/test/modules/test_conv2d.py [hunk 8]
```diff
@@ -1999,12 +2000,24 @@ def test_conv2d_NHWC_with_random_data(test_case):
...
+        torch.nn.functional.conv2d(input, weight_128c, bias=bias, padding=1)
+        torch.nn.functional.conv2d(input, weight_128c, bias=bias, padding=1, stride=2)
+        torch.nn.functional.conv2d(input, weight_1x1_128c, padding=1)
+        torch.nn.functional.conv2d(input, weight_1x1_128c, padding=1, stride=2)
+        torch.nn.functional.conv2d(input, weight_1x1_128c, bias=bias, padding=1)
+        torch.nn.functional.conv2d(input, weight_1x1_128c, bias=bias, padding=1, stride=2)
+        torch.nn.functional.conv2d(input, weight_5x5_128c, padding=1)
+        torch.nn.functional.conv2d(input, weight_5x5_128c, padding=1, stride=2)
+        torch.nn.functional.conv2d(input, weight_5x5_128c, bias=bias, padding=1)
+        torch.nn.functional.conv2d(input, weight_5x5_128c, bias=bias, padding=1, stride=2)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_conv2d.py:2020]: 5x5 卷积的常用 padding 是 2，1x1 卷积常用 padding 是 0

Comment (daquexian) [reply python/oneflow/test/modules/test_conv2d.py:2020]: conv1d 的同理

Comment (laoliu97) [reply python/oneflow/test/modules/test_conv2d.py]: 已修改

**Review by daquexian — APPROVED** [review]

**Review by Flowingsun007 — APPROVED** [review]

File: python/oneflow/test/modules/test_argwhere.py [hunk 9]
```diff
@@ -53,6 +53,10 @@ def test_argwhere_with_random_data(test_case):
         y = torch.argwhere(x)
         return y
 
+    @profile(torch.argwhere)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_argwhere.py:56]: ci 里用的 pytorch 是 1.10，可以在文件头添加 `import torch as torch_original`，并在 @profile 的上一行添加

```python
    @unittest.skipIf(
        version.parse(torch_original.__version__) < version.parse("1.11.0"),
        "torch.argwhere only exists in PyTorch >= 1.11.0",
    )
```
